src/main/java/model/data/process/functions/RowFunction.java

```diff
@@ -0,0 +1,83 @@
+package model.data.process.functions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import model.data.DataRow;
+import model.data.DataTable;
+import model.data.describer.DataDescriber;
+import model.data.value.FloatValue;
+import model.data.value.IntValue;
+import model.data.value.NumberValue;
+
+/**
+ * This class will provide a framework for functions resulting multiple rows of data
+ * @author louisgosschalk
+ *13-05-2015
+ */
+public class RowFunction extends Function{
+	protected boolean minimum;
+	protected boolean maximum;
+	private DataTable table;
+	private DataDescriber<NumberValue> argument;
+	private List<DataRow> rowlist;
+	private DataRow row;
+	
+	public RowFunction(DataTable model, DataDescriber<NumberValue> argument) {
+		super(model, argument);
+		this.table = model;
+		this.argument = argument;
+		this.minimum = false;
+		this.maximum = false;
+		this.rowlist = new ArrayList<DataRow>();
+		this.row = null;
+	}
+	/**
+	 * this function checks restrictions for the calculation and calls its execution
+	 * @return List<DataRow> the rows containing the minimum
+	 */
+	public List<DataRow> calculate() {
+		if(table.getRowCount() == 0)
+			return rowlist; 
+			
+		if(table.getRowCount() == 1) {
+			rowlist.add(table.getRow(0));
+			return rowlist;
+		}
+		row = table.getRow(0);
+		try {
+			argument.resolve(row).getClass();
+		} catch (Exception e) {
+			return rowlist;
+		}
+			if(argument.resolve(row).getClass().equals(FloatValue.class)){
+			return compare();
+		}
+		else if(argument.resolve(row).getClass().equals(IntValue.class)) {
+			return compare();
+		}
+		else {
+			rowlist.clear();
+			return rowlist;
+		}
+	}
+	
+    public List<DataRow> compare() {
+		for(int i = 0; i<table.getRowCount(); i++){
```

Comment:
Bug: in regel 47 zeg je row = table.getRow(0). Je wilt nu kijken of er rijen groter (/kleiner) of als deze rij de grootste is of er rijen gelijk aan deze rij zijn. Maar de eerste rij die je met rij0 vergelijkt is rij0. rij0 is even groot als rij0 en wordt dus aan het resultaat toegevoegd. Dus nu staat rij0 2 keer in het resultaat.